week 2/script.js

```diff
@@ -0,0 +1,219 @@
+    currencyRates.date = new Date().toISOString().split('T')[0];
+    currencyRates.rates[targetCurrency] = rate;
+    document.getElementById('newRateOutput').textContent = JSON.stringify(currencyRates, null, 2);
+  // const select = document.getElementById('targetCurrency').textContent;
+  // const selectCurrencyText = select.options[select.selectedIndex].text;
+   //console.log(currencyRatesArray.findIndex(obj => obj.code == select));
+    currencyRatesArray.push(new CurrencyRate(targetCurrency, rate, currencyRates.date));
+    populateCurrencyDropdown();
+}
+
+function populateCurrencyDropdown() {
+    const selectedFromCurrencyLabel = document.getElementById('selected-from-currency-label');
+    const selectedToCurrencyLabel = document.getElementById('selected-to-currency-label');
+    const selectedCurrencyLabel = document.getElementById('selected-currency-label');
+    const fromCurrency = document.getElementById('fromCurrency');
+    const selectCurrency = document.getElementById('selectCurrency');
+    const toCurrency = document.getElementById('toCurrency');
+    const rates =  currencyRates.rates;
+    
+   /* while (fromCurrency.options.length > 1) {
+        fromCurrency.remove(1);
+        toCurrency.remove(1);
+        selectCurrency.remove(1);
+    }*/
+
+    selectedFromCurrencyLabel.innerHTML = '';
+    selectedToCurrencyLabel.innerHTML = '';
+
+  /*  for (const [key, value] of Object.entries(rates)) {
+        const fromCurrencyOption = document.createElement('option');
+        fromCurrencyOption.value = value;
+        fromCurrencyOption.text = `${key}`;
+        fromCurrency.appendChild(fromCurrencyOption);
+
+        const toCurrencyOption = document.createElement('option');
+        toCurrencyOption.value = value;
+        toCurrencyOption.text = `${key}`;
+        toCurrency.appendChild(toCurrencyOption);
+
+        const selectCurrencyOption = document.createElement('option');
+        selectCurrencyOption.value = value;
+        selectCurrencyOption.text = `${key}`;
+        selectCurrency.appendChild(selectCurrencyOption);
+    }
+
+*/
+
+    fromCurrency.addEventListener('change', function() {
+        selectedFromCurrencyLabel.textContent = fromCurrency.value;
+    });
+
+    toCurrency.addEventListener('change', function() {
+        selectedToCurrencyLabel.textContent = convertRate(findCurrencyRate(fromCurrency.value), findCurrencyRate(toCurrency.value)) + fromCurrency.value;
+    });
+
+    selectCurrency.addEventListener('change', function() {
+        selectedCurrencyLabel.textContent = findCurrencyRate(selectCurrency.value);
+    });
+    console.log(currencyRatesArray);
+    populateTable();
+    populateDatalist();
+}
+
+function findCurrencyRate(currencyCode){
+    let selectedArrIndex = currencyRatesArray.findIndex(obj => obj.code == currencyCode);
+    return currencyRatesArray[selectedArrIndex].rate;
+}
+
+function convertRate(from, to){
+    let rate = (to/from).toFixed(3);
```

Comment:
This conversion works in some cases but does not handle the conversion from the base currency to one of the rates. See the screenshot below. I should be able to convert 100 EUR (the base currency) to DKK; the result should be 12300. But there is an error in the console. Please have a look at this.


<img width="997" alt="image" src="https://github.com/user-attachments/assets/f8a6b604-9922-4935-9691-bb915fce31b4">

Reply:
Done.